pkg/api/dashboard_public.go

```diff
@@ -69,6 +70,30 @@ func (hs *HTTPServer) SavePublicDashboardConfig(c *models.ReqContext) response.R
 	return response.JSON(http.StatusOK, pdc)
 }
 
+// QueryPublicDashboard returns all results for a given panel on a public dashboard
+// POST /api/public/dashboard/:uid/panels/:panelId/query
+func (hs *HTTPServer) QueryPublicDashboard(c *models.ReqContext) response.Response {
+	panelId, err := strconv.ParseInt(web.Params(c.Req)[":panelId"], 10, 64)
+	if err != nil {
+		return response.Error(http.StatusBadRequest, "invalid panel ID", err)
```

Comment:
Let's create proper errors in pkg/models/dashboard_public.go with http status and use handle error function in pkg/api/dashboard_public.go for errors specific to public dashboards